bindings/cpp/CMakeLists.txt

```diff
@@ -0,0 +1,201 @@
+endif()
+
+target_compile_features(${TARGET_NAME} INTERFACE cxx_std_20)
+set_target_properties(${TARGET_NAME} PROPERTIES PUBLIC_HEADER "${SVS_RUNTIME_HEADERS}")
+set_target_properties(${TARGET_NAME} PROPERTIES CXX_STANDARD 20)
+set_target_properties(${TARGET_NAME} PROPERTIES CXX_STANDARD_REQUIRED ON)
+set_target_properties(${TARGET_NAME} PROPERTIES CXX_EXTENSIONS OFF)
+set_target_properties(${TARGET_NAME} PROPERTIES VERSION ${PROJECT_VERSION} SOVERSION ${PROJECT_VERSION_MAJOR} )
+
+if(DEFINED SVS_LVQ_HEADER AND DEFINED SVS_LEANVEC_HEADER)
+    # expected that pre-defined headers are implementation headers
+    message(STATUS "Using pre-defined LVQ header: ${SVS_LVQ_HEADER}")
+    message(STATUS "Using pre-defined LeanVec header: ${SVS_LEANVEC_HEADER}")
+else()
+    set(SVS_LVQ_HEADER "svs/extensions/vamana/lvq.h")
+    set(SVS_LEANVEC_HEADER "svs/extensions/vamana/leanvec.h")
+endif()
+
+if ((SVS_RUNTIME_ENABLE_LVQ_LEANVEC))
+    if(RUNTIME_BINDINGS_PRIVATE_SOURCE_BUILD)
+        message(STATUS "Building directly from private sources")
+        target_link_libraries(${TARGET_NAME} PRIVATE
+            svs::svs
+        )
+        link_mkl_static(${TARGET_NAME})
+    elseif(TARGET svs_static_library)
+        # Links to SVS static library built as part of the main SVS build
+        target_link_libraries(${TARGET_NAME} PRIVATE
+            svs_devel
+            svs_static_library
+            svs_compile_options
+            svs_x86_options_base
+        )
+    elseif(TARGET svs::svs)
+        message(FATAL_ERROR
+            "Pre-built LVQ/LeanVec SVS library cannot be used in SVS main build. "
+            "Please build SVS Runtime using bindins/cpp directory as CMake source root."
+        )
+    else()
+        # Links to LTO-enabled static library, requires GCC/G++ 11.2
+        if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU" AND CMAKE_CXX_COMPILER_VERSION VERSION_GREATER_EQUAL "11.2" AND CMAKE_CXX_COMPILER_VERSION VERSION_LESS "11.3")
+            set(SVS_URL "https://github.com/intel/ScalableVectorSearch/releases/download/v1.0.0-dev/svs-shared-library-1.0.0-NIGHTLY-20251031-740.tar.gz"
+                CACHE STRING "URL to download SVS shared library")
+        else()
+            message(WARNING
+                "Pre-built LVQ/LeanVec SVS library requires GCC/G++ v.11.2 to apply LTO optimizations."
+                "Current compiler: ${CMAKE_CXX_COMPILER_ID} ${CMAKE_CXX_COMPILER_VERSION}"
+            )
+            set(SVS_URL "https://github.com/intel/ScalableVectorSearch/releases/download/v1.0.0-dev/svs-shared-library-1.0.0-NIGHTLY-20251017-faiss.tar.gz")
+        endif()
+        include(FetchContent)
+        FetchContent_Declare(
+            svs
+            URL ${SVS_URL}
+        )
+        FetchContent_MakeAvailable(svs)
+        list(APPEND CMAKE_PREFIX_PATH "${svs_SOURCE_DIR}")
+        find_package(svs REQUIRED)
+        target_link_libraries(${TARGET_NAME} PRIVATE
+            svs::svs
+            svs::svs_compile_options
```

Comment:
I notice when svs::svs_compile_options is not included, performance decreases. I tried it because we do not link it in https://github.com/RedisAI/VectorSimilarity/blob/main/src/VecSim/CMakeLists.txt#L47-L52 - should we?

Reply:
@mihaic, did you manage to check VecSim preformance?

Reply:
I did not but I am tracking that task separately so we can close resolve this conversation.